### DISCUSSION TITLE
Authorization for EC2 instances in AWS

### DISCUSSION THREAD

**Discussion by haimb-rapyd** [istio/istio#50589] [Q&A]
We intend implementing istio in our AWS EKS. We still have some services running on EC2. Are there any options to authorize the requests coming from EC2 based on tags, IAM roles or anything else? I thought also about an agent or proxy that can be installed on EC2 that can communicate with the mesh.

**Comment by rootsongjc** [comment]
You have several ways to integrate EC2 instances with Istio on AWS EKS:

## **Ingress Gateway**

Use Istio's built-in gateway to manage traffic from EC2 to EKS. This approach is straightforward and uses Istio's capabilities to secure and monitor traffic entering the Kubernetes cluster from EC2.

## **VPN/Direct Connect**

Connect your EC2 instances to your Kubernetes network using AWS VPN or Direct Connect. This makes your EC2 instances part of the same network as your EKS cluster, allowing you to manage them more seamlessly within Istio. However, the cost of doing so will be relatively high, and it is not worth it.

## **Istio on EC2**

Install Istio directly on your EC2 instance, see [the doc](https://istio.io/latest/docs/setup/install/virtual-machine/). This involves setting up the same Istio sidecar proxies that you use within EKS on your EC2 instances, enabling uniform management and security policies across your cloud and on-premises resources. But at present, this step is more complicated, there is no open source tool to help you achieve, you can only manually operate. Tetrate's TSB, however, can run on both virtual machines and Kubernetes, helping you establish a connection between them. See this blog [Mobile Premier League Migrates from VMs to K8s in AWS with Tetrate to Deliver a Better Experience to 90+ Million Gamers](https://tetrate.io/blog/vm-onboarding-with-tsb-service-mesh-and-customer-experience/) and TSB doc [Migrating VM Monoliths to your cluster](https://docs.tetrate.io/service-bridge/howto/traffic/migrating-VM-monoliths) and [configure the VM](https://docs.tetrate.io/service-bridge/setup/workload-onboarding/quickstart/aws-ec2/configure-vm#install-workload-onboarding-agent).
